src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+import java.util.Scanner;
 public class Duke {
+    public static void printLine() {
+        System.out.println("____________________________________________________________");
+    }
+
+    //list tasks
+    public static void listTasks(Task[] list, int count){
+        printLine();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 1; i <= count; i++) {
+            System.out.println(i + "." + list[i - 1]);
+        }
+        printLine();
+    }
+
+    //add a new task
+    public static void addTask(Task[] list, int count, String command){
+        Task t = new Task(command);
+        list[count] = t;
+        //count++;
+        printLine();
+        System.out.println("added: " + t.description); //new item added
+        printLine();
+    }
+
+    //mark task as done
+    public static void markTask(Task[] list, String[] input, int i){
+        int index = Integer.parseInt(input[i]);
+        Task t = list[index-1];
+        t.markAsDone();
+        printLine();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(t);
+        printLine();
+    }
+
+    //mark task as not done
+    public static void unmarkTask(Task[] list, String[] input, int i){
+        int index = Integer.parseInt(input[i]);
+        Task t = list[index-1];
+        t.markNotDone();
+        printLine();
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println(t);
+        printLine();
+    }
+
+    //to-do
+    public static void addTodo(Task[] list, int count, String[] input, int i){
+        Task t = new Todo(input[i]);
+        list[count] = t;
+        count++;
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println(t);
+        System.out.println("Now you have " + count + " tasks in the list.");
+        printLine();
+    }
+
+    //deadline
+    public static void addDeadline(Task[] list, int count, String[] input, int i, int j){
+        Task t = new Deadline(input[i], input[j]);
+        list[count] = t;
+        count++;
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println(t);
+        System.out.println("Now you have " + count + " tasks in the list.");
+        printLine();
+    }
+
+    //event
+    public static void addEvent(Task[] list, int count, String[] start, String[] end, int i, int j, int k){
```

Comment:
Too many function parameters, perhaps could find a way to simplify it by abstracting it to different functions instead?